lms/static/scripts/frontend_apps/components/dashboard/StudentsActivity.tsx

```diff
@@ -6,45 +6,48 @@ import {
 } from '@hypothesis/frontend-shared';
 import type { DataTableProps } from '@hypothesis/frontend-shared';
 import { useState } from 'preact/hooks';
+import { useParams } from 'wouter-preact';
 
-import type { StudentStats } from '../../api-types';
+import type { Assignment, StudentStats } from '../../api-types';
+import { useConfig } from '../../config';
+import { useAPIFetch } from '../../utils/api';
 import { formatDateTime } from '../../utils/date';
 
-export type AssignmentInfo = {
-  title: string;
-};
-
-export type StudentsActivityTableProps = {
-  assignment: AssignmentInfo;
-  students: StudentStats[];
-  loading?: boolean;
-};
-
 type MandatoryOrder<T> = NonNullable<DataTableProps<T>['order']>;
 
-export default function StudentsActivityTable({
-  assignment,
-  students,
-  loading,
-}: StudentsActivityTableProps) {
-  const title = `Assignment: ${assignment.title}`;
+export default function StudentsActivity() {
+  const { dashboard } = useConfig(['dashboard']);
+  const { routes } = dashboard;
+  const { assignmentId } = useParams<{ assignmentId: string }>();
+  const assignment = useAPIFetch<Assignment>(
+    routes.assignment.replace(':assignment_id', assignmentId),
+  );
+  const students = useAPIFetch<StudentStats[]>(
+    routes.assignment_stats.replace(':assignment_id', assignmentId),
+  );
+
+  const title = `Assignment: ${assignment.data?.title}`;
   const [order, setOrder] = useState<MandatoryOrder<StudentStats>>({
     field: 'display_name',
     direction: 'ascending',
   });
-  const orderedStudents = useOrderedRows(students, order);
+  const orderedStudents = useOrderedRows(students.data ?? [], order);
 
   return (
     <Card>
       <CardContent>
         <h2 className="text-brand mb-3 text-xl" data-testid="title">
-          {title}
+          {assignment.isLoading && 'Loading...'}
+          {assignment.error && 'Could not load assignment title'}
+          {assignment.data && title}
```

Comment:
I reckon this is not an awesome approach, but I guess it should do for now, until we add something a bit better with `DataLoader` which combines loading the assignment data with the list of students.

Reply:
We need to figure out the visual design first and then we can decide how to implement it. There is also a retry mechanism in `useAPIFetch` which we could look at integrating.